elasticsearch/init/0200-seed-index-templates

```diff
@@ -17,6 +17,25 @@
 
 source "logging"
 
+info Deleting deprecated index templates
+shopt -s failglob
+for deprecated_file in ${ES_HOME}/index_templates/deprecated/*.json
```

Comment:
didn't we talk about just deleting any templates that match our naming pattern that we don't provide in the es image anymore?

```
templates = getListOfTemplates()
for template in templates
  if template !matchProvidedTemplates
    delete template
  fi
done
```

Reply:
@ewolinetz I tried that but it was making the script more complicated with multiple loops. Also, we have to embed python to parse the output of `_cat/templates`. I feel this would be a better way of handling by putting in deprecated folder and then deleting. This also reduces little overhead and code looks cleaner to me.

Reply:
@ewolinetz Do we not allow CUs to push in their own index templates?

Reply:
> @ewolinetz Do we not allow CUs to push in their own index templates?

we can, but we can just advise them to not match our template naming convention

> @ewolinetz I tried that but it was making the script more complicated with multiple loops. Also, we have to embed python to parse the output of `_cat/templates`.

Why do we need python to parse the output? the `_cat` end points print new line delimited strings which feeds into a loop easily.

Reply:
also, it doesn't make sense to maintain a list of deprecated templates simply to delete them from the cluster. for the sanity of the repo we should remove ones that we don't provide that match our naming convention.

Reply:
@sasagarw I think what @ewolinetz suggests is the following.

Get two lists:

1) **First list** (what comes in the image)
- pull index templates files from provided folder (where files from https://github.com/openshift/origin-aggregated-logging/tree/master/elasticsearch/index_templates are copied)
- filter out all files that do not match `common.settings.*.template.json` or `com.redhat.*.template.json`
2) **Second list** (what is currently in ES cluster)
- do the same, except pull index templates from ES, ideally from `_cat/templates` (you can [tell it to output only a single column](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/cat.html#headers) with the name of the index template - which is the same as the file name it was created from)
- again filter out all items that do not match the above mentioned patters

Delete from ES all index templates that are found in list 2) but not in list 1).

Reply:
yes, except we don't need to filter for the first list :)

Reply:
@ewolinetz @lukas-vlcek both the ways we will be deleting custom templates created by CUs. Also, we create `ocp-gen-app` and `ocp-gen-infra` templates which will not match the pattern and will be deleted wrongly. 
For `ocp-gen-(app|infra)` I can filter using pattern `ocp-gen-*` but the custom templates of CUs will be wrongly deleted.